### PR TITLE
Return a hash for a given library

### LINKED ISSUE #20335: Create create to generate hashes for our module naming convention
## Reason
Need a concentric way of producing hash based on several files/directories which will be used to name our moose-petsc/moose-libmesh modules:

```pre
moose-petsc/abc123
```

## Design
Generate a hash for any files/directories involved with building said module. Then, produce a final hash based on those results.

```bash
usage: module_hash.py [-h] library [library ...]

Supplies a hash for a given library based on MOOSE HEAD.

positional arguments:
  library     choose from: petsc, libmesh

optional arguments:
  -h, --help  show this help message and exit
```

## Impact
A generic, easily controlled way to produce a module name hash

### PR DESCRIPTION
**Pull request by milljm**
A simple script to produce a hash based on several key files involved when
building the vetted libraries; PETSc and libMesh.

Closes #20335

### CODE REVIEW

**Review by loganharbour — CHANGES_REQUESTED** [review]
Add a second positional argument `ref` whose default is `HEAD`?

That way we can do `./module_hash.py libmesh <some commit>` without having to check it out

It would also allow us to add a unit test

File: scripts/module_hash.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                           os.path.join(root_path, submodule)],
+                          stdout=subprocess.PIPE,
+                          stderr=subprocess.PIPE,
+                          text=True,
+                          shell=False) as proc:
+        proc.wait()
+        if proc.poll() == 0:
+            return proc.communicate(0)[0].split()[2]
+        print(proc.communicate(1))
+        sys.exit(1)
```
Comment (loganharbour) [inline scripts/module_hash.py:63]:
```suggestion
    out = subprocess.check_output(['git', 'ls-tree', 'HEAD', os.path.join(root_path, submodule)]).decode('utf-8')
    return out.split()[2]
```

Comment (milljm) [reply scripts/module_hash.py:63]: Do we not want to handle an error?

Comment (loganharbour) [reply scripts/module_hash.py]: If I remember right, check output prints stderr to screen and throws an exception on a nonzero return code

**Review by loganharbour — DISMISSED** [review]
```
ERRORS:
/var/civet2/build/moose/scripts/tests/tests:module_hash:37
No 'design' supplied
Control this message using: 'log_missing_design' 
```

Looks like we need to create a website markdown entry for this utility 😆

**Comment by permcody** [comment]
I realize I'm really late to the party, but some of this logic is already in the `get_repo_hash.py` script that's in `framework/scripts` that we've been using for a long time to get version information.

**Comment by loganharbour** [comment]
Some of it is but the purpose is very different.

**Comment by milljm** [comment]
Is there a reason we can't merge this yet?

**Review by loganharbour — APPROVED** [review]

**Pull request merged**